presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -163,6 +168,25 @@ public void checkCanShowTablesMetadata(ConnectorTransactionHandle transactionHan
         return tableNames;
     }
 
+    @Override
+    public List<ColumnMetadata> filterColumns(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, CatalogSchemaTableName tableName, List<ColumnMetadata> columns)
+    {
+        if (!checkTablePermission(transactionHandle, identity, tableName.getSchemaTableName(), SELECT, false)) {
```

Comment:
I looked at https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization#SQLStandardBasedHiveAuthorization-Privileges
It says that "SHOW COLUMNS" requires "SELECT" privilege on the table.
Please let me know in case I'm misunderstanding the spec.

Reply:
I was referring to the SQL specification.  My memory is visibility is attached to having any permissions.  For example, you can have `INSERT` permission without `SELECT` permissions.

Reply:
Changed the implementation to check for any table permission